### DISCUSSION TITLE
Welcome to the Carbon Aware SDK

### DISCUSSION THREAD

**Discussion by vaughanknight** [Green-Software-Foundation/discussions#65] [Carbon Aware SDK]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

## Discussion vs Issues or the Wiki?
**Issues** will be specifically used to create backlog aligned with the goals, raise bugs, and ensure we have strong tracking of the work being delivered with clear purpose.

The **Wiki** will be used to document things relating to the project, contributions, core tenets, and process.  Carbon Aware SDK documentation will live in the main repository.

To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by vaughanknight** [comment]
I should introduce myself.  I'm Vaughan Knight, and I work as a Principal Engineering Manager at Microsoft, based in Sydney Australia.  

I manage an awesome group fo engineering teams in the Asia time zone, and work with teams all around the world, with a focus on Manufacturing & Energy customers.  

I'm very passionate about how we can incorporate Green Software principles into software engineering with low or no friction, and finding ways to transform how software is operates easily, for both new software, and legacy software.  

I believe the Carbon Aware SDK project is the first step to unlocking huge potential to reduce emissions across a wide array of software and platforms, and we are yet to fully comprehend the scale of this impact.

**Comment by srini1978** [comment]
I am Srini and I work as a program manager in the Azure Identity Group within Microsoft, based out of Hyderabad India. 

I have a background in Software architecture and I have been working on both Cloud and on-premise architecture solutions for customers (both enterprise and small business). I have worked with Enterprise and small business customers for architecting Web  and API solutions in Azure using PaaS services, containers, Kubernetes and Relational database products. 

I am passionate about Sustainability as a vision and even more passionate about building and helping build assets that can help developers and customers plan , architect and build sustainable software. I have been a regular writer on sustainable software engineering within Microsoft and have written quite a few blogs

**Reply by srini1978** [reply]
> I am Srini and I work as a program manager in the Azure Identity Group within Microsoft, based out of Hyderabad India.
> 
> I have a background in Software architecture and I have been working on both Cloud and on-premise architecture solutions for customers (both enterprise and small business). I have worked with Enterprise and small business customers for architecting Web and API solutions in Azure using PaaS services, containers, Kubernetes and Relational database products.
> 
> I am passionate about Sustainability as a vision and even more passionate about building and helping build assets that can help developers and customers plan , architect and build sustainable software. I have been a regular writer on sustainable software engineering within Microsoft and have written quite a few blogs

I probably missed mentioning the areas where i can contribute

1) I have background in .Net development and can help there for the Carbon aware SDK
2) I also have experience in architecting solutions and hence I can help in the architecture decisions and decisions around how we want to build a solution that is sustainable ourselves
3) I have experience in writing blogposts and documentation

**Comment by Henry-WattTime** [comment]
Hello All, Henry Richardson here. I'm an analyst at WattTime, a non-profit focused on providing grid emissions factors to partners to achieve impact. I think the SDK will be a key tool to enabling this!

**Comment by will-iamalpine** [comment]
Hi everyone! I'm a program manager on Microsoft's Azure Machine team. I drove the 'carbon-aware API' which is the predecessor to the [carbon-aware SDK](https://github.com/TaylorPrewitt/carbon-awareAPI), and is being used internally by several Microsoft teams. Can't wait to extract whatever logic is useful and make this fully open and interoperable!

**Comment by fcarvalhodev** [comment]
Hello everyone, I'm a C# and .Net software developer from Thoughtworks/Brazil. I have experience developing embbed software, web api's, desktop applications and mobile, but what really make me happy, it's develop projects that make sense such as Carbon aware SDK.

I hope that our work here may change the software developer scenario, and I cant wait to start to help! :)

**Comment by luwangshell** [comment]
Hi there! This is Lu Wang from Shell Software Engineering group. I have been working in oil and gas industry for about 15 years, and have experience on C# (winforms, wpf/mvvm in the old days, SOLID design principles, .net core, etc). Excited to see this initiative and digital solution can help shape the future of O&G industry!

**Comment by ghost** [comment]
Hello all, I am Rakesh Siri from Accenture India. I have experience in Data Science and Data Engineering with 5+ years of experience.  Looking forward to working with you all and making the world less carbon crowded!

**Comment by hectortosa** [comment]
Hi everyone! 👋 

I'm Héctor Tortosa from Spain. I work as a Lead Consultant Developer and Tech Lead in Thoughtworks Spain. Currently, I'm involved in the retail sector, but previously I have worked in media, telecom, and enterprise software (ERPs). I have experience doing SDKs, web APIs, and cloud architecture.

I'm very passionate about climate change and all things related. I hope that one day, energy efficiency and carbon footprint will be part of the standard development process like today is security, performance, etc

**Comment by will-iamalpine** [comment]
Hi everyone!

I'm Will Buchanan from Microsoft. I work as a program manager (TPM) at
Microsoft. Much of my work involves carbon awareness; I prototyped a
carbon-aware
API <http://azure-uw-cli-2021.azurewebsites.net/ci_data> which has been
adopted by some internal Microsoft teams. I hope we can fold this work into
the carbon-aware SDK.

Additionally, I founded and lead the GreenAI community at Microsoft, and
recently launched some exciting features
<https://techcommunity.microsoft.com/t5/green-tech-blog/charting-the-path-towards-sustainable-ai-with-azure-machine/ba-p/2866923>
to show the computational cost of ML workloads. The goal is for ML to
become carbon-aware on Azure....so we need an SDK to make this happen :)

Best,
Will

**Comment by nikhilsamant** [comment]
Hello everyone, 
I'm Nikhil and I work as Senior Consultant in Thoughtworks India. 

I have experience working on Cloud architecture solutions and have a good background working in C# and .NET. I have worked with clients for architecting and developing Web Apps and Web APIs. I have also worked on few on-premise to cloud migration projects.

I have a goal of my own to use energy efficiently, reduce my carbon emission and contributing to GSF is what I feel, one of the ways of moving towards this goal.

Looking forward to working with this project.

**Comment by andynolantw** [comment]
Hi everyone,
I'm Andy Nolan, Director of Emerging Technology at Thoughtworks based in Melbourne Australia. 

We're developing our green software engineering sensible defaults to inform how we design systems and help our clients. 

This project will lower the complexity for developers looking to build carbon aware system. It's a much needed tool. 

Thanks!

**Comment by joecus1** [comment]
Hi Everyone,

I'm Joe Cusano. I work as a software developer at Glenview Capital Management in New York city. I am new to sustainable development / green software, but eager to learn more about it. I have experience building web apps with C# / .NET, and would love to contribute to the CASDK.

Looking forward to learning more and helping out!
